### PR TITLE
Build KubeOne container image

### LINKED ISSUE #1868: Build KubeOne container image
**What feature would you like to be added?**

We should build and publish a container image for KubeOne.

**What are use cases of the feature?**

* Run KubeOne in a container
* https://github.com/kubermatic/kubermatic/issues/8950

### PR DESCRIPTION
**Pull request by xmudrii**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

This PR introduces an experimental KubeOne container image. The image is:

* built using `buildah` for `amd64` and `arm64`
* available in the `quay.io/kubermatic/kubeone` repository, which is already created.
* based on `alpine:3.15` with the `openssh-client` package installed (so we can use SSH and ssh-agent).

This PR also adds the `pull-kubeone-build-image` job, which tests building the image on each PR. This is very fast (around 2 minutes) because of the recent introduction of `arm64` GOCACHE (#1876).

The image is pushed on each commit to master and on each release.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1868

**Does this PR introduce a user-facing change?**:
```release-note
[EXPERIMENTAL] Add the KubeOne container image. This image should NOT be used in the production.
```

### CODE REVIEW

**Comment by xmudrii** [comment]
/assign @kron4eg @ahmedwaleedmalik 
cc @imharshita

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ENV GOCACHE=$GOCACHE
+
+WORKDIR /go/src/k8c.io/kubeone
+COPY . .
+RUN make build
+
+FROM docker.io/alpine:3.15
+LABEL maintainer="support@kubermatic.com"
+
+RUN apk add --update openssh-client
```
Comment (kron4eg) [inline Dockerfile:30]: for ssh agent?

Comment (xmudrii) [reply Dockerfile:30]: @kron4eg Yes, exactly, I mentioned this in the PR description.

Comment (xmudrii) [reply Dockerfile:30]: And not only ssh-agent, but also for the `ssh` binary itself (it's not included by default).

Comment (xmudrii) [reply Dockerfile]: Added a comment to the Dockerfile about this.

[the remark this answers is not in the document]
Comment (xmudrii) [reply Dockerfile]: @kron4eg Fixed, PTAL.

**Review by kron4eg — APPROVED** [review]

**Pull request merged**